src/code/z_demo.c

```diff
@@ -756,35 +784,36 @@ void Cutscene_Command_ChooseCreditsScenes(PlayState* play, CutsceneContext* csCt
                 }
                 play->transitionTrigger = TRANS_TRIGGER_START;
                 break;
+
+            default:
+                break;
         }
     }
 }
 
-// Command 0x99: Motion blur
-void Cutscene_Command_MotionBlur(PlayState* play, CutsceneContext* csCtx, CsCmdBase* cmd) {
-    if ((csCtx->frames >= cmd->startFrame) && (cmd->endFrame >= csCtx->frames)) {
-        if ((csCtx->frames == cmd->startFrame) && (cmd->base == 1)) {
+void CutsceneCmd_MotionBlur(PlayState* play, CutsceneContext* csCtx, CsCmdMotionBlur* cmd) {
+    if ((csCtx->curFrame >= cmd->startFrame) && (cmd->endFrame >= csCtx->curFrame)) {
+        if ((csCtx->curFrame == cmd->startFrame) && (cmd->type == CS_MOTION_BLUR_ENABLE)) {
             Play_EnableMotionBlur(180);
         }
 
-        if (cmd->base == 2) {
-            f32 progress = Environment_LerpWeight(cmd->endFrame, cmd->startFrame, csCtx->frames);
+        if (cmd->type == CS_MOTION_BLUR_DISABLE) {
+            f32 lerp = Environment_LerpWeight(cmd->endFrame, cmd->startFrame, csCtx->curFrame);
 
-            if (progress >= 0.9f) {
+            if (lerp >= 0.9f) {
                 Play_DisableMotionBlur();
             } else {
-                Play_SetMotionBlurAlpha((1.0f - progress) * 180.0f);
+                Play_SetMotionBlurAlpha((1.0f - lerp) * 180.0f);
             }
         }
     }
 }
 
-// Command 0x9A: Gives Tatl to the player
-void Cutscene_Command_GiveTatlToPlayer(PlayState* play, CutsceneContext* csCtx, CsCmdBase* cmd) {
+void CutsceneCmd_GiveTatlToPlayer(PlayState* play, CutsceneContext* csCtx, CsCmdGiveTatl* cmd) {
     Player* player = GET_PLAYER(play);
 
-    if (csCtx->frames == cmd->startFrame) {
-        if (cmd->base == 1) {
+    if (csCtx->curFrame == cmd->startFrame) {
+        if (cmd->giveTatl == true) {
             gSaveContext.save.hasTatl = true;
             if (player->tatlActor != NULL) {
                 return;
```

Comment:
not your change but == NULL and no return seems probably better? 
i think it makes sense to say "if tatl is missing, spawn her"

Reply:
That does make much more sense. Unfortunately, it actually seems like it breaks matching lmao. Which is probably why it ended up like this to begin with